readme.md

```diff
@@ -185,6 +362,10 @@ Then, go to [http://localhost:5000/](http://localhost:5000) and you should see a
 
 * **Your Answer:**
 
+Middleware is not run if a route above it matches.
+
+It could be at the top of the server pipeline, eg for logging.  Authentication could be used for only certain routes, if use is put in-between routes.
+
```

Comment:
Middleware is the code/application/module that communicates between two systems, such as between servers and databases.